skimage/restoration/deconvolution.py

```diff
@@ -364,7 +364,9 @@ def richardson_lucy(image, psf, num_iter=50, clip=True, filter_epsilon=None):
     Parameters
     ----------
     image : ndarray
-       Input degraded image (can be n-dimensional).
+       Input degraded image (can be n-dimensional). Input image
+       should be normalized to the range [0, 1] to allow for proper
+       floating point conversion prior to deconvolution.
```

Comment:
Hmm, this function is a bit weird. It seems like normalization is only necessary because the clipping defaults to -1 and 1. I don't see anything in the deconvolution algorithm itself that suggests normalization is necessary. But I may be missing something.

Also the clipping suggest that the normalization range can also be [-1, 1]?

@scikit-image/core, is this scaling behavior we want to update for skimage2? I don't have a lot of context on this API.

Reply:
@lagru I admit I am also not super familiar with the API; the reason I decided to open this PR was that testing this function on images represented as float64 `np.ndarray`s with no particular constraints on their values (iirc the images I used as inputs were between ~20 and ~80) resulted in blank images.
In retrospect, the docstring of the `clip` parameter explains the observed outcome, nonetheless I think some kind of warning in the main docstring is helpful.
I can test the function again with different inputs and come up with better wording if you agree.

Reply:
Yes, please do. Right now it might be more confusing that clarifying to users. 

Coming back to this, I think we should use the planned skiamge2 API update to switch to `clip=False` as the new default.

Reply:
I haven't looked at the code recently, but my guess would be that clipping is implemented in restoration routines because results can so easily blow up. E.g., if you aren't careful while applying inverse Fourier domain filters, you often divide by very small numbers:

https://github.com/scikit-image/scikit-image/blob/main/skimage/restoration/deconvolution.py#L92

Clipping saves you from very large values creeping in, but also makes the operation highly non-linear. So... not sure what the default should be, but turning clipping off should at least be an option, if it's not the default.

@mkcor That `wiener` docstring is really not well written, in case you have a moment to take a look.

Reply:
Hi @geeanlooca,

Thank you for bringing this to our attention.

> It seems like normalization is only necessary because the clipping defaults to -1 and 1. I don't see anything in the deconvolution algorithm itself that suggests normalization is necessary.

@lagru absolutely right.

Yesterday we held a sprint to make decisions about the transition to `skimage2`. We discussed this issue, that current restoration functions clip values by default, and clip them to [-1, 1]. Our conclusion is that this is something we want to change in `skimage2`: By default, we won't clip values; we will ask the user to provide their data range (max - min values of the input image) and, if they want clipping (which is typically the case, as @stefanv explained above), we will clip based on the data range.

So, for now, let's stick with a docstring improvement, indeed. I'll make a suggestion to mention clipping.